cli/azd/pkg/project/project_config_test.go

```diff
@@ -428,3 +429,61 @@ services:
 
 	require.Equal(t, "hello", projectConfig.ResourceGroupName.MustEnvsubst(env.Getenv))
 }
+
+func TestMinVersion(t *testing.T) {
+	savedVersion := internal.Version
+	t.Cleanup(func() {
+		internal.Version = savedVersion
+	})
+
+	const testProjWithVersion = `
+name: test-proj
+metadata:
+  template: test-proj-template
+  minSupportedVersion: 0.6.0-beta.3
+`
+
+	const testProjWithoutVersion = `
+name: test-proj
+metadata:
+  template: test-proj-template
+`
+
+	t.Run("noVersion", func(t *testing.T) {
+		internal.Version = "0.6.0-beta.3 (commit 0000000000000000000000000000000000000000)"
+
+		_, err := ParseProjectConfig(testProjWithoutVersion)
+		require.NoError(t, err)
+	})
+
+	t.Run("supportedVersion", func(t *testing.T) {
+		// Exact match of minimum version.
+		internal.Version = "0.6.0-beta.3 (commit 0000000000000000000000000000000000000000)"
+
+		_, err := ParseProjectConfig(testProjWithVersion)
+		require.NoError(t, err)
+
+		// Newer version than minimum.
+		internal.Version = "0.6.0 (commit 0000000000000000000000000000000000000000)"
+
+		_, err = ParseProjectConfig(testProjWithVersion)
+		require.NoError(t, err)
+	})
+
+	t.Run("unsupportedVersion", func(t *testing.T) {
+		internal.Version = "0.6.0-beta.2 (commit 0000000000000000000000000000000000000000)"
```

Comment:
Do we need to add special handling for `daily` builds until we stop shipping with `beta`? Let's add a test for this behavior.